File: src/charting/highlight/RadarHighlighter.java
package com.github.mikephil.charting.highlight;

import com.github.mikephil.charting.charts.RadarChart;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.interfaces.datasets.IDataSet;
import com.github.mikephil.charting.utils.MPPointF;
import com.github.mikephil.charting.utils.Utils;

import java.util.List;

/**
 * Created by philipp on 12/06/16.
 */
public class RadarHighlighter extends PieRadarHighlighter<RadarChart> {

    public RadarHighlighter(RadarChart chart) {
        super(chart);
    }

    
    protected Highlight getClosestHighlight(let index, let x, let y) {

        List<Highlight> highlights = getHighlightsAtIndex(index);

        let distanceToCenter = this.mChart.distanceToCenter(x, y) / this.mChart.getFactor();

        Highlight closest = null;
        let distance = Number.MAX_VALUE;

        for (let i = 0; i < highlights.length; i++) {

            Highlight high = highlights.get(i);

            let cdistance = Math.abs(high.getY() - distanceToCenter);
            if (cdistance < distance) {
                closest = high;
                distance = cdistance;
            }
        }

        return closest;
    }
    /**
     * Returns an array of Highlight objects for the given index. The Highlight
     * objects give information about the value at the selected index and the
     * DataSet it belongs to. INFORMATION: This method does calculations at
     * runtime. Do not over-use in performance critical situations.
     *
     * @param index
     * @return
     */
    protected List<Highlight> getHighlightsAtIndex(let index) {

        this.mHighlightBuffer.clear();

        let phaseX = this.mChart.getAnimator().getPhaseX();
        let phaseY = this.mChart.getAnimator().getPhaseY();
        let sliceangle = this.mChart.getSliceAngle();
        let factor = this.mChart.getFactor();

        MPPointF pOut = MPPointF.getInstance(0,0);
        for (let i = 0; i < this.mChart.getData().getDataSetCount(); i++) {

            IDataSet<?> dataSet = this.mChart.getData().getDataSetByIndex(i);

            final Entry entry = dataSet.getEntryForIndex(index);

            let y = (entry.getY() - this.mChart.getYChartMin());

            Utils.getPosition(
                    this.mChart.getCenterOffsets(), y * factor * phaseY,
                    sliceangle * index * phaseX + this.mChart.getRotationAngle(), pOut);

            this.mHighlightBuffer.add(new Highlight(index, entry.getY(), pOut.x, pOut.y, i, dataSet.getAxisDependency()));
        }

        return this.mHighlightBuffer;
    }
}

File: src/charting/highlight/RadarHighlighter.ts
import { RadarChart } from '../charts/RadarChart';
import { RadarDataSet } from '../data/RadarDataSet';
import { RadarEntry } from '../data/RadarEntry';
import { MPPointF } from '../utils/MPPointF';
import { Utils } from '../utils/Utils';
import { PieRadarHighlighter } from './PieRadarHighlighter';

/**
 * Created by philipp on 12/06/16.
 */
export class RadarHighlighter extends PieRadarHighlighter<RadarEntry, RadarDataSet, RadarChart> {
    protected getClosestHighlight(index: number, x: number, y: number) {
        const highlights = this.getHighlightsAtIndex(index);

        const distanceToCenter = this.mChart.distanceToCenter(x, y) / this.mChart.getFactor();

        let closest = null;
        let distance = Infinity;

        for (let i = 0; i < highlights.length; i++) {
            const high = highlights[i];
            // const set = this.mChart.getData().getDataSetByIndex(high.dataSetIndex);
            // const yProperty = set.yProperty;
            const cdistance = Math.abs(high.y - distanceToCenter);
            if (cdistance < distance) {
                closest = high;
                distance = cdistance;
            }
        }

        return closest;
    }
    /**
     * Returns an array of Highlight objects for the given index. The Highlight
     * objects give information about the value at the selected index and the
     * DataSet it belongs to. INFORMATION: This method does calculations at
     * runtime. Do not over-use in performance critical situations.
     *
     * @param index
     * @return
     */
    protected getHighlightsAtIndex(index) {
        this.mHighlightBuffer = [];

        const phaseX = this.mChart.getAnimator().getPhaseX();
        const phaseY = this.mChart.getAnimator().getPhaseY();
        const sliceangle = this.mChart.getSliceAngle();
        const factor = this.mChart.getFactor();

        const pOut: MPPointF = { x: 0, y: 0 };
        for (let i = 0; i < this.mChart.getData().getDataSetCount(); i++) {
            const dataSet = this.mChart.getData().getDataSetByIndex(i);
            const yKey = dataSet.yProperty;

            const entry = dataSet.getEntryForIndex(index);

            const y = entry[yKey] - this.mChart.getYChartMin();

            Utils.getPosition(this.mChart.getCenterOffsets(), y * factor * phaseY, sliceangle * index * phaseX + this.mChart.getRotationAngle(), pOut);

            this.mHighlightBuffer.push({
                entry,
                x: index,
                y: entry[yKey],
                xPx: pOut.x,
                yPx: pOut.y,
                dataSetIndex: i,
                axis: dataSet.getAxisDependency(),
            });
        }

        return this.mHighlightBuffer;
    }
}
